### PR TITLE
Improve Adreno conv2d_nchwc

### PR DESCRIPTION
**Pull request by echuraev**
* Change Adreno conv2d schedule to generate more valid kernles

* fix a case of unfriendly sizes for split which can get local work size in [32, 1024]

* Fix of tune/eval kernel divergence

* Move call of multi_filter after latest knob definition

* Expand conv having odd numbers to be dividable by 2 and 4

  Required for inception v3 having prime numbers in spacial dimentions in the shape

* Fix for shape missmatch after extension of odd spatial dims

* Add tests on nchwc texture conv

* Fixed pad for texture cache_read

* Add dirty fix of AutoTVM for config having multi filter

* Fix memory markup for adreno in case of scalar const for some inputs

* Vectorize postops for conv2d

### CODE REVIEW

File: tvm/python/tvm/autotvm/task/space.py [hunk 1]
```diff
@@ -822,6 +826,33 @@ def valid(self):
         """
         return not bool(self.errors)
 
+    def multi_filter(self, **kwargs):
```
Comment (TejashShah) [inline tvm/python/tvm/autotvm/task/space.py:829]: Is this multi_filter added to generate valid kernel configs that won't result in kernel crashes/reboot?

https://github.com/octoml/qualcomm/blob/125ea01298a195e4366100b00f4d71297a1e605e/tvm/python/tvm/topi/adreno/conv2d.py#L218

File: tvm/python/tvm/te/tensor.py [hunk 2]
```diff
@@ -84,12 +84,6 @@ def __eq__(self, other):
             if isinstance(other, _expr.ExprOp):
                 return _expr.EqualOp(self, other)
             return False
-        if self.ndim == 0 and other.ndim == 0:
```
Comment (TejashShah) [inline tvm/python/tvm/te/tensor.py:87]: Are we sure that this removal doesn't have any side effects?

File: tvm/python/tvm/autotvm/task/space.py (hunk 1 above)
Comment (elvin-n) [reply tvm/python/tvm/autotvm/task/space.py:829]: yes, we decrease search space by adding multifilter for all axis of local work group in mul must not exceed 1024.

File: tvm/python/tvm/te/tensor.py (hunk 2 above)
Comment (elvin-n) [reply tvm/python/tvm/te/tensor.py:87]: I cannot recalд exact network, either yolv3 or deeplab has const scalar tensors which tried to be converted into textures and we need to remove this verification. By fact this verificatin prohibited of comparing tensor rank-0. If we remove this verification, everything starts to work properly

Comment (TejashShah) [reply tvm/python/tvm/te/tensor.py:87]: Thanks.

**Comment by TejashShah** [comment]
LGTM

cc @csullivan @tmoreau89

**Pull request merged**